src/psyclone/tests/psyad/tl2ad_test.py

```diff
@@ -419,16 +428,114 @@ def test_generate_adjoint_multi_kernel(fortran_reader):
         "    real :: psyir_tmp, psyir_tmp_1\n"
         "    psyir_tmp = psyir_tmp_1\n"
         "  end subroutine kern2\n"
-        "  subroutine kern3()\n"
+        "end module test_mod\n")
+    expected_ad_fortran_str = (
+        "module adj_test_mod\n"
+        "  implicit none\n"
+        "  public\n\n"
+        "  contains\n"
+        "  subroutine adj_kern1()\n"
+        "    real :: psyir_tmp\n"
+        "    real :: psyir_tmp_1\n\n"
+        "    psyir_tmp = 0.0\n"
+        "    psyir_tmp_1 = 0.0\n"
+        "    psyir_tmp_1 = psyir_tmp_1 + psyir_tmp\n"
+        "    psyir_tmp = 0.0\n\n"
+        "  end subroutine adj_kern1\n"
+        "  subroutine adj_kern2()\n"
+        "    real :: psyir_tmp\n"
+        "    real :: psyir_tmp_1\n\n"
+        "    psyir_tmp = 0.0\n"
+        "    psyir_tmp_1 = 0.0\n"
+        "    psyir_tmp_1 = psyir_tmp_1 + psyir_tmp\n"
+        "    psyir_tmp = 0.0\n\n"
+        "  end subroutine adj_kern2\n\n"
+        "end module adj_test_mod\n")
+    psyir = fortran_reader.psyir_from_source(tl_fortran_str)
+    ad_psyir = generate_adjoint(psyir, ["psyir_tmp", "psyir_tmp_1"])
+    ad_fortran_str = fortran_writer(ad_psyir)
+    assert ad_fortran_str == expected_ad_fortran_str
+    assert Compile(tmpdir).string_compiles(ad_fortran_str)
+
+
+def test_generate_adjoint_no_module(fortran_reader, fortran_writer, tmpdir):
```

Comment:
I'm a bit confused here - has this PR also altered PSyAD functionality? I thought it was just adding the kernels.